entities/weapons/raft_builder/cl_init.lua

```diff
@@ -267,4 +341,37 @@ function SWEP.drawItemRequirement( x, y, itemType, requirement, font )
 
     surface.SetTextPos( x - w * 0.5 + iconSize + 5, y - textH * 0.5 )
     surface.DrawText( text )
+
+    return count >= requirement
+end
+
+function SWEP:UpdatePermitted()
+    self.permitted = true
+
+    local trace = self:GetOwner():GetEyeTrace()
+    local ent = trace.Entity
+    local isRaftEnt = IsValid( ent ) and ent.IsRaft
+    if not ( isRaftEnt and ent:GetRaft():CanBuild( LocalPlayer() ) ) then
+        self.permitted = false
+        return
+    end
+
+    local mins, maxs = self.ghost:GetModelBounds()
+    local pos = self.ghost:GetPos()
+
+    local traceData = {
+        start = pos,
+        endpos = pos,
+        filter = self.ghost,
+        mins = mins + Vector( 5, 5, 5 ),
+        maxs = maxs + Vector( -5, -5, 5 ),
+        mask = MASK_ALL,
+        ignoreworld = true
+    }
+
+    local traceResult = util.TraceHull( traceData )
```

Comment:
should collisions checks be done serverside as well?

Reply:
Added a TODO, not that important and couldn't work it out at first attempt